### PR TITLE
Translation: Getting Started & Developers & Technology & Learn [zh]

### PR DESCRIPTION
**Pull request by Junweif2**
There are four parts of documentation. We first submit Getting Started & Developers for team to review, in case there are some formats issues. 
If there are no fatal error, then we will adjust the other half and submit another PR.

### CODE REVIEW

**Comment by Junweif2** [comment]
Submitted "Translation: Technology & Learn[zh]" for team to review, thank you all

Also open a issue about a small type "https://github.com/scroll-tech/scroll-documentation/issues/119"

File: src/content/docs/zh/developers/ethereum-and-scroll-differences.mdx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+| Opcode                      | Solidity equivalent | Scroll Behavior                                                                                            |
+| --------------------------- | ------------------- | ---------------------------------------------------------------------------------------------------------- |
+| `BLOCKHASH`                 | `block.blockhash`   | 返回最新256个区块的 `keccak(chain_id \|\| block_number)`                                     |
+| `COINBASE`                  | `block.coinbase`    | 返回预部署的费用合约地址，查看 [Scroll 合约](/developers/scroll-contracts). |
+| `DIFFICULTY` / `PREVRANDAO` | `block.difficulty`  | 返回s 0.                                                                                                 |
+| `BASEFEE`                   | `block.basefee`     | 禁用。[^eip1559] 如果引入该操作码，交易将被还原                       |
+| `SELFDESTRUCT`              | `selfdestruct`      | 禁用。如果引入该操作码，交易将被还原。[^willadpot]                      |
+
+[^eip1559]: We have currently disabled EIP-1559 on Scroll.
```
Comment (ChuhanJin) [inline src/content/docs/zh/developers/ethereum-and-scroll-differences.mdx]: seems like this can be also translated to Chinese

File: src/content/docs/zh/developers/ethereum-and-scroll-differences.mdx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  对于一般的 Solidity 开发者来说，这些细节不会影响你的开发体验。
+</Aside>
+
+## EVM 操作码
+
+| Opcode                      | Solidity equivalent | Scroll Behavior                                                                                            |
+| --------------------------- | ------------------- | ---------------------------------------------------------------------------------------------------------- |
+| `BLOCKHASH`                 | `block.blockhash`   | 返回最新256个区块的 `keccak(chain_id \|\| block_number)`                                     |
+| `COINBASE`                  | `block.coinbase`    | 返回预部署的费用合约地址，查看 [Scroll 合约](/developers/scroll-contracts). |
+| `DIFFICULTY` / `PREVRANDAO` | `block.difficulty`  | 返回s 0.                                                                                                 |
```
Comment (ChuhanJin) [inline src/content/docs/zh/developers/ethereum-and-scroll-differences.mdx]: no s

File: src/content/docs/zh/developers/guides/bridge-erc20-through-the-custom-gateway.mdx [hunk 3]
```diff
@@ -0,0 +1,355 @@
...
+    IL2ScrollMessenger(messenger).sendMessage{ value: msg.value }(counterpart, 0, _message, _gasLimit);
+
+    emit WithdrawERC20(_l1Token, _token, _from, _to, _amount, _data);
+  }
+}
+```
+
+### 在 Sepolia 上设置 Gateway 合约
+
+部署合约后，调用以下函数初始化合约，并将其绑定到相应的代币和跨链桥另一端的网关。
```
Comment (ChuhanJin) [inline src/content/docs/zh/developers/guides/bridge-erc20-through-the-custom-gateway.mdx]: 网关 can replace to gateway, feel better and make more sense

File: src/content/docs/zh/developers/guides/estimating-gas-and-tx-fees.mdx [hunk 4]
```diff
@@ -0,0 +1,227 @@
+---
```
Comment (ChuhanJin) [inline src/content/docs/zh/developers/guides/estimating-gas-and-tx-fees.mdx:1]: no translation?

File: src/content/docs/zh/developers/guides/scroll-messenger-cross-chain-interaction.mdx [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+date: Last Modified
+title: "Scroll Messenger 跨链交互"
+lang: "zh"
+permalink: "developers/guides/scroll-messenger-cross-chain-interaction"
+excerpt: "In this example, we will launch a dummy smart contract on either Sepolia or Scroll Sepolia testnet and interact with it from the opposite chain."
+---
+
+import Aside from "../../../../../components/Aside.astro"
+
+在这个例子中，我们将在Sepolia或Scroll上启动一个虚拟智能合约，并从对方的链上与之交互。我们将使用同时部署在Sepolia和Scroll上的 `ScrollMessenger` 合约。
```
Comment (ChuhanJin) [inline src/content/docs/zh/developers/guides/scroll-messenger-cross-chain-interaction.mdx]: Scroll -> Scroll Sepolia

File: src/content/docs/zh/developers/guides/scroll-messenger-cross-chain-interaction.mdx [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+      gasLimit,
+      msg.sender
+    );
+  }
+}
+```
+
+## 调用跨链方法
+
+We pass the message by executing `executeFunctionCrosschain` and passing the following parameters:
```
Comment (ChuhanJin) [inline src/content/docs/zh/developers/guides/scroll-messenger-cross-chain-interaction.mdx]: delete this line?

File: src/content/docs/zh/developers/index.mdx [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+<ToggleElement anchor="community">
+  <div slot="title">社区——Scroll 将用户和开发者聚集在一起。</div>
+  <p>
+    我们知道在主网发布之前进行开源开发并获得用户参与的挑战！Scroll 拥有一个蓬勃发展的用户和开发者社区，并且拥有超过 100,000 名渴望在我们的测试网上试用应用程序的 Discord 社区，我们很高兴将开发者与可以提供真实世界反馈的用户联系起来。
+  </p>
+</ToggleElement>
+
+## 感谢你与我们一起构建开发
+
+我们正在努力为网络带来更多的集成和支持更多的基础设施，并为我们未来的主网版本感到兴奋。
```
Comment (ChuhanJin) [inline src/content/docs/zh/developers/index.mdx:74]: check the status, because we already launched our mainnet
"We’re working to bring more integrations and support infrastructure to the network, and are excitedly working towards our Mainnet release."
this sentence need to be updated also?
@dghelm @Turupawn

File: src/content/docs/zh/developers/l1-and-l2-bridging/enforced-transactions.mdx [hunk 8]
```diff
@@ -0,0 +1,57 @@
+---
```
Comment (ChuhanJin) [inline src/content/docs/zh/developers/l1-and-l2-bridging/enforced-transactions.mdx:1]: translate or not?

File: src/content/docs/zh/developers/l1-and-l2-bridging/the-scroll-messenger.mdx [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+
+## L1 交易最终确认
+
+任何从L2发起的交易都需要使用Scroll Messenger合约上的 `relayMessageWithProof` 方法最终确认。我们将此过程称为“提交执行提款交易”，并且通过网关或路由器发送任意消息和转移资产都需要它。当您使用 `relayMessageWithProof` 时，您必须提供默克尔树包含证明，证明您的交易包含在“提款”消息的树中，以及其他参数。生成此证明和这些值可以在本地无需许可的情况下完成，但目前，检索这些参数的最简单方法是通过我们托管在 https://sepolia-api-bridge.scroll.io/api/ 的后端 API。
+
+<Aside type="danger" title="实验性 API">
+  此 API 是为我们的跨链桥 UI 所定制。它尚未最终确定，将来可能会发生变化。当 API 最终确定时，我们将更新本指南。
+</Aside>
+
+Supply the address of the EOA or contract responsible for initiating the original transaction on L2 to the `/claimable`
```
Comment (ChuhanJin) [inline src/content/docs/zh/developers/l1-and-l2-bridging/the-scroll-messenger.mdx]: delete?

File: src/content/docs/zh/developers/transaction-fees-on-scroll.mdx [hunk 10]
```diff
@@ -0,0 +1,174 @@
...
+
+import Aside from "../../../../components/Aside.astro"
+
+## 概览
+
+Scroll Sepolia测试网的费用明显低于其底层。对于用户和开发者来说，Scroll上的交易费用看上去与以太坊主网上的交易费用类似，现有的工具，钱包和代码可能会像以前一样工作。但是，钱包中显示的交易费用并不是全部，除非软件特别了解Scroll的费用计算。
+
+由于 L2 rollups的设计，交易成本取决于 L1 的成本。为了利用以太坊的安全性，Scroll必须考虑必须在L1上存储和验证交易数据和证明的成本。
+
+Compared to Ethereum mainnet, Scroll Sepolia introduces some new dimensions to transaction fee calculation to do this. The final cost of a transaction is constructed from several parts:
```
Comment (ChuhanJin) [inline src/content/docs/zh/developers/transaction-fees-on-scroll.mdx]: delete?

File: src/content/docs/zh/developers/transaction-fees-on-scroll.mdx [hunk 11]
```diff
@@ -0,0 +1,174 @@
...
+```
+总费用取决于交易gas使用情况（ `l2TransactionGasUsed` ）以及当前的市场状况（ `l2TransactionGasPrice` ）。用户设置 gas 价格，并通过在 Scroll 节点上调用 `estimateGas` 端口来预估“使用的gas”。
+
+换句话说，执行费用的计算方式与[EIP1559](https://eips.ethereum.org/EIPS/eip-1559)之前的以太坊完全相同。
+
+## L1 费用
+
+每笔交易的调用数据（calldata）必须提交到以太坊，这会产生额外的交易费用，称为“L1费用”。如果不这样做，则无法仅从 L1 数据重新恢复 Scroll。
+
+Transactions aren't committed 1-by-1 -- they are collected in batches of blocks (and blocks of transactions). The cost of an individual transaction is computed based on the zeros and non-zero bytes in its payload.
```
Comment (ChuhanJin) [inline src/content/docs/zh/developers/transaction-fees-on-scroll.mdx]: delete?

File: src/content/docs/zh/developers/transaction-fees-on-scroll.mdx [hunk 12]
```diff
@@ -0,0 +1,174 @@
...
+
+Scroll 预先部署了 L1GasPriceOracle 合约 ，可在 Scroll Sepolia 上访问([`0x5300000000000000000000000000000000000002`](https://sepolia-blockscout.scroll.io/address/0x5300000000000000000000000000000000000002))。它提供了 `getL1Fee` 的方法，可以估算给定交易原始数据的 L1 数据费用。
+
+```solidity
+function getL1Fee(bytes memory _data) external view override returns (uint256);
+```
+
+<Aside type="note" title="如果 L1 gas 价格飙升会发生什么?">
+
+Once the sequencer has processed a transaction, a user's L1 fee is locked in, and any fluctuations will not affect what a user pays.
```
Comment (ChuhanJin) [inline src/content/docs/zh/developers/transaction-fees-on-scroll.mdx]: delete?

File: src/content/docs/zh/getting-started/overview.md [hunk 13]
```diff
@@ -0,0 +1,43 @@
...
+
+- 访问我们的 [跨链桥](https://scroll.io/bridge) 并连接您的钱包
+- 将代币从 Sepolia 发送到 Scroll Sepolia 测试网 (或者使用 [水龙头](/user-guide/faucet))
+- 测试我们的 [Uniswap 用例](http://uniswap-showcase.sepolia.scroll.xyz/) -- 只需要确保选择了 Scroll Sepolia 网络!
+
+<!-- TODO: Add Aave after Sepolia goes live -->
+
+## Scroll 的未来方向?
+
+在以太坊上启动主网之前，我们目前处于最终测试网中。接下来将去中心化堆栈的每个组件。查看 [我们的路线图](https://scroll.io) 或者关注 [我们的 Discord](https://discord.gg/scroll) 或者 [Twitter](https://twitter.com/scroll_zkp)。
```
Comment (ChuhanJin) [inline src/content/docs/zh/getting-started/overview.md]: “We’ve released our mainnet on Ethereum, but there’s still more work to do. Next up — decentralizing each component of the stack. To stay up to date, check out [our blog](https://scroll.io/blog) or follow along in [our Discord](https://discord.gg/scroll) or on [Twitter](https://twitter.com/scroll_zkp).”
should we update this ? @dghelm @Turupawn

File: src/content/docs/zh/technology/bridge/withdraw-gateways.mdx [hunk 14]
```diff
@@ -0,0 +1,173 @@
...
+title: "提款网关"
+lang: "zh"
+permalink: "technology/bridge/withdraw-gateways"
+whatsnext: { "排序器": "/technology/sequencer/execution-node/" }
+---
+
+import ClickToZoom from "../../../../../components/ClickToZoom.astro"
+import WithdrawWorkflow from "../_images/withdraw.png"
+
+This document describes how users and developers can utilize gateways to withdraw tokens from L2 to L1. We provide several gateways for standard tokens and a gateway router on L2, listed in the table below.
```
Comment (ChuhanJin) [inline src/content/docs/zh/technology/bridge/withdraw-gateways.mdx]: delete?

File: src/content/docs/zh/technology/chain/blocks.mdx [hunk 15]
```diff
@@ -0,0 +1,40 @@
...
+lang: "zh"
+permalink: "technology/chain/blocks"
+whatsnext: { "Rollup 流程": "/technology/chain/rollup/" }
+---
+
+A block is a collection of transactions and links to the previous block it extends.
+
+## 区块头
+
+The block header in Scroll mirrors the structure of Ethereum's. However, certain fields in the block header have been modified to accommodate the Layer 2 context, as detailed in the following table:
```
Comment (ChuhanJin) [inline src/content/docs/zh/technology/chain/blocks.mdx]: delete?

File: src/content/docs/zh/technology/chain/blocks.mdx [hunk 16]
```diff
@@ -0,0 +1,40 @@
+---
+section: technology
+date: Last Modified
+title: "区块"
+lang: "zh"
+permalink: "technology/chain/blocks"
+whatsnext: { "Rollup 流程": "/technology/chain/rollup/" }
+---
+
+A block is a collection of transactions and links to the previous block it extends.
```
Comment (ChuhanJin) [inline src/content/docs/zh/technology/chain/blocks.mdx]: why no translation here

File: src/content/docs/zh/getting-started/overview.md (hunk 13 above)
Comment (Junweif2) [reply src/content/docs/zh/getting-started/overview.md]: Resolve the sentence in Chinese

File: src/content/docs/zh/developers/l1-and-l2-bridging/enforced-transactions.mdx (hunk 8 above)
Comment (Junweif2) [reply src/content/docs/zh/developers/l1-and-l2-bridging/enforced-transactions.mdx:1]: Same as "estimating-gas-and-tx-fees.mdx", I didn't see this page show up in local development and online docs in English. And I check the Spanish version, they also remains untranslated. 
I guess it's a page working in progress?

File: src/content/docs/zh/developers/guides/estimating-gas-and-tx-fees.mdx (hunk 4 above)
Comment (Junweif2) [reply src/content/docs/zh/developers/guides/estimating-gas-and-tx-fees.mdx:1]: I didn't see this page show up in local development and online docs in English. And I check the Spanish version, they also remains untransalted.
I guess it's a page working in progress?

File: src/content/docs/zh/developers/index.mdx (hunk 7 above)
Comment (Junweif2) [reply src/content/docs/zh/developers/index.mdx:74]: Resolve the sentence in Chinese

File: src/content/docs/zh/developers/index.mdx [hunk 17]
```diff
@@ -71,6 +71,6 @@ Scroll通过执行链下交易以及生成交易正确执行的加密证明来
 
 ## 感谢你与我们一起构建开发
 
-我们正在努力为网络带来更多的集成和支持更多的基础设施，并为我们未来的主网版本感到兴奋。
+我们正在努力为 Scroll 主网带来更多的集成和支持更多的基础设施。
```
Comment (ChuhanJin) [inline src/content/docs/zh/developers/index.mdx]: can revert this change, since EN version is not ready

File: src/content/docs/zh/getting-started/overview.md [hunk 18]
```diff
@@ -40,4 +40,4 @@ Scroll 的 Sepolia 测试网已经上线。 虽然我们有一个 [完整指南]
 
 ## Scroll 的未来方向?
 
-在以太坊上启动主网之前，我们目前处于最终测试网中。接下来将去中心化堆栈的每个组件。查看 [我们的路线图](https://scroll.io) 或者关注 [我们的 Discord](https://discord.gg/scroll) 或者 [Twitter](https://twitter.com/scroll_zkp)。
+Scroll 目前已经上线以太坊上的主网，接下来将去中心化堆栈的每个组件。查看 [我们的路线图](https://scroll.io) 或者关注 [我们的 Discord](https://discord.gg/scroll) 或者 [Twitter](https://twitter.com/scroll_zkp)。
```
Comment (ChuhanJin) [inline src/content/docs/zh/getting-started/overview.md]: revert this change ditto

Comment (Junweif2) [reply src/content/docs/zh/getting-started/overview.md]: done

File: src/content/docs/zh/developers/index.mdx (hunk 17 above)
Comment (Junweif2) [reply src/content/docs/zh/developers/index.mdx]: done

**Review by ChuhanJin — DISMISSED** [review]

**Review by ChuhanJin — DISMISSED** [review]

**Review by ChuhanJin — DISMISSED** [review]

**Review by ChuhanJin — APPROVED** [review]
